WalletWasabi/WabiSabi/Client/AmountDecomposer.cs

```diff
@@ -280,12 +296,29 @@ public IEnumerable<Money> Decompose(IEnumerable<Money> myInputCoinEffectiveValue
 					count,
 					stdDenoms).Select(Money.Satoshis).ToList();
 
+				// Translate back to denominations.
+				List<Output> finalDenoms = new();
+				foreach (var outputPlusFee in currentSet)
+				{
+					finalDenoms.Add(denoms.First(d => d.EffectiveCost == outputPlusFee));
+				}
+
+				// The decomposer won't take vsize into account for different script types, checking it back here if too much, disregard the decomposition.
+				var totalVSize = finalDenoms.Sum(d => d.ScriptType.EstimateOutputVsize());
+				if (totalVSize > AvailableVsize)
+				{
+					continue;
+				}
+
 				hash = new();
-				foreach (var item in currentSet.OrderBy(x => x))
+				foreach (var item in finalDenoms.OrderBy(x => x.EffectiveCost))
 				{
-					hash.Add(item);
+					hash.Add(item.Amount);
 				}
-				setCandidates.TryAdd(hash.ToHashCode(), (currentSet, myInputSum - (ulong)currentSet.Sum() + (ulong)count * OutputFee + (ulong)count * InputFee)); // The cost is the remaining + output cost + input cost.
+
+				var deficit = (myInputSum - (ulong)finalDenoms.Sum(d => d.EffectiveCost)) + CalculateCostMetrics(finalDenoms);
```

Comment:
This thing is never used anywhere.

Reply:
![YouAreFakeNews-FakeNewsGIF](https://user-images.githubusercontent.com/9844978/219639899-607b7cbd-92c7-4e52-8ec1-826fd69cd406.gif)

In the next line.

Reply:
ohh `deficit` is `Cost`, ok.